userspace/libsinsp/filterchecks.cpp

```diff
@@ -109,6 +110,317 @@ static inline bool str_match_start(const std::string& val, size_t len, const cha
+
+	m_path_checks.emplace(PPME_SYSCALL_FCHOWN_X, evt_fd_name);
+	m_success_checks.emplace(PPME_SYSCALL_FCHOWN_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_FCHOWNAT_X, evt_arg_pathname);
+	m_success_checks.emplace(PPME_SYSCALL_FCHOWNAT_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_QUOTACTL_X, evt_arg_special);
+	m_success_checks.emplace(PPME_SYSCALL_QUOTACTL_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_RENAME_X, evt_arg_oldpath);
+	m_target_checks.emplace(PPME_SYSCALL_RENAME_X, evt_arg_newpath);
+	m_success_checks.emplace(PPME_SYSCALL_RENAME_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_RENAMEAT_X, evt_arg_oldpath);
+	m_target_checks.emplace(PPME_SYSCALL_RENAMEAT_X, evt_arg_newpath);
+	m_success_checks.emplace(PPME_SYSCALL_RENAMEAT_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_RENAMEAT2_X, evt_arg_oldpath);
+	m_target_checks.emplace(PPME_SYSCALL_RENAMEAT2_X, evt_arg_newpath);
+	m_success_checks.emplace(PPME_SYSCALL_RENAMEAT2_X, evt_arg_res_eq_0);
+
+	m_success_checks.emplace(PPME_SYSCALL_LINK_X, evt_arg_res_eq_0);
+
+	m_success_checks.emplace(PPME_SYSCALL_LINKAT_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_LINK_2_X, evt_arg_newpath);
+	m_target_checks.emplace(PPME_SYSCALL_LINK_2_X, evt_arg_oldpath);
+	m_success_checks.emplace(PPME_SYSCALL_LINK_2_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_LINKAT_2_X, evt_arg_newpath);
+	m_target_checks.emplace(PPME_SYSCALL_LINKAT_2_X, evt_arg_oldpath);
+	m_success_checks.emplace(PPME_SYSCALL_LINKAT_2_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_SYMLINK_X, evt_arg_linkpath);
+	m_target_checks.emplace(PPME_SYSCALL_SYMLINK_X, evt_arg_target);
+	m_success_checks.emplace(PPME_SYSCALL_SYMLINK_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_SYMLINKAT_X, evt_arg_linkpath);
+	m_target_checks.emplace(PPME_SYSCALL_SYMLINKAT_X, evt_arg_target);
+	m_success_checks.emplace(PPME_SYSCALL_SYMLINKAT_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_MOUNT_X, evt_arg_dev);
+	m_target_checks.emplace(PPME_SYSCALL_MOUNT_X, evt_arg_dir);
+	m_success_checks.emplace(PPME_SYSCALL_MOUNT_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_UMOUNT_X, evt_arg_name);
+	m_success_checks.emplace(PPME_SYSCALL_UMOUNT_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_UMOUNT_1_X, evt_arg_name);
+	m_success_checks.emplace(PPME_SYSCALL_UMOUNT_1_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_UMOUNT2_X, evt_arg_name);
+	m_success_checks.emplace(PPME_SYSCALL_UMOUNT2_X, evt_arg_res_eq_0);
+}
+
+sinsp_filter_check* sinsp_filter_check_fspath::allocate_new()
+{
+	return (sinsp_filter_check*)new sinsp_filter_check_fspath();
+}
+
+uint8_t* sinsp_filter_check_fspath::extract(sinsp_evt* evt, OUT uint32_t* len, bool sanitize_strings)
+{
+	*len = 0;
+	ASSERT(evt);
+
+	m_tstr = "";
+
+	if(m_path_checks.empty())
```

Comment:
Any compelling reason for doing this lazily instead of at init time?

Reply:
I guess not. What I should really do is make them static so there aren't ~15 additional filterchecks created for every new fs.path.* filtercheck. and now that I'm going to get the factory out of the way, this is possible. So I'll do that.

Reply:
Hrm, I can't make them static as sinsp_filter_check_event and sinsp_filter_check_fd both use m_inspector to check the buffer format. But I will make the set of filterchecks shared, tied to the inspector.